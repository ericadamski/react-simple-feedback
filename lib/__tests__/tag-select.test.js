import React from "react";
import { render, fireEvent, cleanup } from "react-testing-library";
import TagSelect from "../index";

const data = {
  tags: [
    {
      label: "Earn More",
      emoji: "💵",
      value: "earn more"
    },
    {
      label: "Change Jobs",
      emoji: "💼",
      value: "change jobs"
    },
    {
      label: "Work Remotely",
      emoji: "🏠",
      value: "work remotely"
    },
    {
      label: "Build an Idea",
      emoji: "🔧",
      value: "build an idea"
    },
    {
      label: "Something Else",
      emoji: "🤔",
      value: "something else"
    }
  ],
  message: "Thank you for your feedback 👍",
  label: "Eric's goal is to..."
};

describe("Tag Select", () => {
  beforeEach(cleanup);

  it("should match the snapshot", () => {
    /**
     * default render
     */
    expect(render(<TagSelect {...data} />).container).toMatchSnapshot();
    /**
     * default complete render
     */
    expect(
      render(<TagSelect {...data} complete />).container
    ).toMatchSnapshot();
  });

  it("should toggle on click of a tag", () => {
    const onClick = jest.fn();

    const { getByText } = render(<TagSelect {...data} onClick={onClick} />);

    fireEvent.click(getByText(data.tags[0].label).parentNode);

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith(data.tags[0].value);
  });

  it("should call onClose on click of the close button", () => {
    const close = jest.fn();

    const { getByTestId } = render(<TagSelect {...data} onClose={close} />);

    fireEvent.click(getByTestId("close-button"));

    expect(close).toHaveBeenCalledTimes(1);
  });

  it("should set a timeout and clear that timeout on close click", () => {
    const timeout = setTimeout;
    const cleartimeout = clearTimeout;
    const close = jest.fn();

    global.setTimeout = jest.fn(fn => (fn(), "called"));
    global.clearTimeout = jest.fn();

    const { getByTestId, getByText } = render(
      <TagSelect {...data} hideOnComplete hideTimeout={1000} onClose={close} />
    );

    fireEvent.click(getByText(data.tags[0].label).parentNode);

    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 1000);

    expect(close).toHaveBeenCalled();

    fireEvent.click(getByTestId("close-button"));

    expect(clearTimeout).toHaveBeenCalledTimes(1);
    expect(clearTimeout).toHaveBeenCalledWith("called");

    /**
     * Just want to put these back to make sure they don't get forgotten
     */
    setTimeout = timeout;
    clearTimeout = cleartimeout;
  });
});
