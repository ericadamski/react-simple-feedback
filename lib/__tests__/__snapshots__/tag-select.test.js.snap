// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Select should match the snapshot 1`] = `
<div>
  <div
    class="sc-iwsKbI gmMYAg"
  >
    <p
      class="sc-gzVnrw hMxoia"
    >
      Thank you for your feedback 👍
    </p>
    <div
      class="sc-bZQynM ijsBjn sc-EHOje ewpEue"
      data-testid="close-button"
      role="button"
    />
    <p
      class="sc-dnqmqq flsexd"
    >
      Eric's goal is to...
    </p>
    <ul
      class="sc-htoDjs bZzhDu sc-bwzfXH kJsiIi"
    >
      <li
        class="sc-bdVaJa bDWFJH"
      >
        <div
          class="sc-htpNat GPaEE"
        >
          <span
            class="sc-bxivhb fWJver"
            role="img"
          >
            💵
          </span>
          <p
            class="sc-ifAKCX jHATIr"
          >
            Earn More
          </p>
        </div>
      </li>
      <li
        class="sc-bdVaJa bDWFJH"
      >
        <div
          class="sc-htpNat GPaEE"
        >
          <span
            class="sc-bxivhb fWJver"
            role="img"
          >
            💼
          </span>
          <p
            class="sc-ifAKCX jHATIr"
          >
            Change Jobs
          </p>
        </div>
      </li>
      <li
        class="sc-bdVaJa bDWFJH"
      >
        <div
          class="sc-htpNat GPaEE"
        >
          <span
            class="sc-bxivhb fWJver"
            role="img"
          >
            🏠
          </span>
          <p
            class="sc-ifAKCX jHATIr"
          >
            Work Remotely
          </p>
        </div>
      </li>
      <li
        class="sc-bdVaJa bDWFJH"
      >
        <div
          class="sc-htpNat GPaEE"
        >
          <span
            class="sc-bxivhb fWJver"
            role="img"
          >
            🔧
          </span>
          <p
            class="sc-ifAKCX jHATIr"
          >
            Build an Idea
          </p>
        </div>
      </li>
      <li
        class="sc-bdVaJa bDWFJH"
      >
        <div
          class="sc-htpNat GPaEE"
        >
          <span
            class="sc-bxivhb fWJver"
            role="img"
          >
            🤔
          </span>
          <p
            class="sc-ifAKCX jHATIr"
          >
            Something Else
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Tag Select should match the snapshot 2`] = `
<div>
  <div
    class="sc-iwsKbI dtCEkz"
  >
    <p
      class="sc-gzVnrw hMxoia"
    >
      Thank you for your feedback 👍
    </p>
    <div
      class="sc-bZQynM ijsBjn sc-EHOje ewpEue"
      data-testid="close-button"
      role="button"
    />
    <p
      class="sc-dnqmqq flsexd"
    >
      Eric's goal is to...
    </p>
    <ul
      class="sc-htoDjs bZzhDu sc-bwzfXH kJsiIi"
    >
      <li
        class="sc-bdVaJa bDWFJH"
      >
        <div
          class="sc-htpNat GPaEE"
        >
          <span
            class="sc-bxivhb fWJver"
            role="img"
          >
            💵
          </span>
          <p
            class="sc-ifAKCX jHATIr"
          >
            Earn More
          </p>
        </div>
      </li>
      <li
        class="sc-bdVaJa bDWFJH"
      >
        <div
          class="sc-htpNat GPaEE"
        >
          <span
            class="sc-bxivhb fWJver"
            role="img"
          >
            💼
          </span>
          <p
            class="sc-ifAKCX jHATIr"
          >
            Change Jobs
          </p>
        </div>
      </li>
      <li
        class="sc-bdVaJa bDWFJH"
      >
        <div
          class="sc-htpNat GPaEE"
        >
          <span
            class="sc-bxivhb fWJver"
            role="img"
          >
            🏠
          </span>
          <p
            class="sc-ifAKCX jHATIr"
          >
            Work Remotely
          </p>
        </div>
      </li>
      <li
        class="sc-bdVaJa bDWFJH"
      >
        <div
          class="sc-htpNat GPaEE"
        >
          <span
            class="sc-bxivhb fWJver"
            role="img"
          >
            🔧
          </span>
          <p
            class="sc-ifAKCX jHATIr"
          >
            Build an Idea
          </p>
        </div>
      </li>
      <li
        class="sc-bdVaJa bDWFJH"
      >
        <div
          class="sc-htpNat GPaEE"
        >
          <span
            class="sc-bxivhb fWJver"
            role="img"
          >
            🤔
          </span>
          <p
            class="sc-ifAKCX jHATIr"
          >
            Something Else
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
